http.go

```diff
@@ -22,33 +22,57 @@ type (
 		UserAgent string
 		headers   *http.Header
 	}
+
+	// HTTPSourceClientOptFunc set some values to client
+	HTTPSourceClientOptFunc func(*HTTPSourceClient) error
 )
 
-// NewHTTPSourceClient create HTTPSourceClient object
-func NewHTTPSourceClient(url *url.URL, category, hostname, sourceName *string) *HTTPSourceClient {
-	return &HTTPSourceClient{
-		url:       url,
-		client:    &http.Client{},
-		UserAgent: UserAgent(),
-		headers:   getHeaders(category, hostname, sourceName),
+// SetXSumoCategoryHeader set X-Sumo-Category's value to header
+func SetXSumoCategoryHeader(category string) HTTPSourceClientOptFunc {
+	return func(h *HTTPSourceClient) error {
+		if category != "" {
+			h.headers.Add("X-Sumo-Category", category)
+		}
+		return nil
 	}
 }
 
-func getHeaders(category *string, hostname *string, sourceName *string) *http.Header {
-	if category == nil && hostname == nil && sourceName == nil {
+// SetXSumoNameHeader set X-Sumo-Name's value to header
+func SetXSumoNameHeader(name string) HTTPSourceClientOptFunc {
+	return func(h *HTTPSourceClient) error {
+		if name != "" {
+			h.headers.Add("X-Sumo-Name", name)
```

Comment:
minor comment: maybe we can use and extender version of mergeHeaders (that considers "merged" can be null) rather than assuming HTTPSourceClient arrives with headers != nil in all opts functions?